app/server/lib/FlexServer.ts

```diff
@@ -1987,6 +1994,21 @@ export class FlexServer implements GristServer {
     this.create.addExtraHomeEndpoints(this, this.app);
   }
 
+  public async startCheckingForUpdates() {
+    const gristServer = this;
+    await compareWithLatest(gristServer);
+    this._intervalCheckID = setInterval(async () => { await compareWithLatest(gristServer); }, ONE_WEEK);
```

Comment:
Things started by `setTimeout` and `setInterval` must be diposed properly when the server is closed, take a look in the `Housekeeper.ts`. It has a close method where it stops all intervals from happening.

Reply:
Thanks, that makes sense. I've moved all of this logic to `Housekeeper.ts`.